homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -1308,6 +1311,9 @@ class PipelineInput:
     stt_stream: AsyncIterable[bytes] | None = None
     """Input audio for stt. Required when start_stage = stt."""
 
+    stt_wake_up_key: str | None = None
+    """Optional key used to de-duplicate wake-ups when start_stage = stt."""
```

Comment:
So I am a bit confused here. I thought the goal was that we would try to de-duplicate across both local and Wyoming-based wake words.

- Why is it called `stt_wake_up_key` and not `wake_word_phrase` or just `wake_word` ? 
- Why limit to `start_stage = stt`? That would not work with different styles of wake words.

Reply:
Maybe I'm being too pedantic by calling it a "wake-up key" :smile:  But my reasoning was that some wake word models support multiple phrases like "ok, jarvis" and "hey, jarvis". We don't technically know the wake phrase that the user said, just whatever was in the manifest (under `wake_word`).

The different styles (streaming vs local) is a problem. Right now, the wake word system + wake word id combo is used as the wake-up key. I think we should change this to the wake word name, which should be something like "ok nabu" instead of `ok_nabu_v0.1` and do some normalization so "hey_jarvis" and "Hey Jarvis" still match.

Reply:
We can make rules in our manifest how they should specify wake words. Maybe the property `wake_word_phrase` is the most correct here. A single wake word engine could match on multiple wake words.

Reply:
That makes sense. Do you think we should alter the streaming wake word check too? It seems like we're moving towards just checking if the "wake phrases" (reasonably) match.

Reply:
What do you mean with streaming wake word ?

Reply:
Say you have 2 satellites in the same room, one with local wake word and one using a streaming wake word. Both use "ok, nabu".

Right now, saying "ok, nabu" would wake them both up and duplicate the voice command. Streaming wake word uses the wake word system + model name to deconflict ("openWakeWord.ok_nabu") whereas local would just use the wake phrase ("ok nabu").

I'm proposing these should be considered identical.

Reply:
I agree. We should just look at ok nabu.

Reply:
Using just the wake word phrase now.